extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftDeployer.java

```diff
@@ -24,11 +24,17 @@ public void checkEnvironment(Optional<SelectedKubernetesDeploymentTargetBuildIte
                 return;
             }
             if (target.getEntry().getName().equals(OPENSHIFT)) {
-                if (client.getClient().isAdaptable(OpenShiftClient.class)) {
-                    deploymentCluster.produce(new KubernetesDeploymentClusterBuildItem(OPENSHIFT));
-                } else {
-                    throw new IllegalStateException(
-                            "Openshift was requested as a deployment, but the target cluster is not an Openshift cluster!");
+                try (var openShiftClient = client.getClient().adapt(OpenShiftClient.class)) {
+                    if (openShiftClient.isSupported()) {
+                        deploymentCluster.produce(new KubernetesDeploymentClusterBuildItem(OPENSHIFT));
+                    } else {
+                        throw new IllegalStateException(
+                                "Openshift was requested as a deployment, but the target cluster is not an Openshift cluster!");
+                    }
+                } catch (Exception e) {
+                    throw new RuntimeException(
+                            "Failed to adapt 'OpenShiftClient', please add 'io.fabric8:openshift-client' to your dependencies: ",
```

Comment:
I don't think we encourage users to use non-Quarkus dependencies to make extensions work. 
When could this exception be thrown? Maybe when users use the `quarkus-kubernetes` extension and select `openshift` as the deployment target? If so, better to let users use the `quarkus-openshift` extension. Moreover, the sentence "Failed to adapt `OpenShiftClient`" is a bit technical, better to write a comprehensible and general message: "Failed to configure OpenShift. Make sure you have the Quarkus OpenShift extension.".

Reply:
Sure, that makes much more sense. Done.